### PR TITLE
copy

### PR DESCRIPTION
**Pull request by ayushkamat**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -238,7 +238,7 @@ task buildInfo { doLast {
  * Compile sources and copies all libs to target directory
  */
 task compile {
-    dependsOn buildInfo
+//    dependsOn buildInfo // ayush: remove this to stop all the Const.groovy merge conflicts
```
Comment (rahuldesai1) [inline build.gradle:241]: thank you so much

File: modules/nf-commons/src/main/nextflow/file/CopyMoveHelper.java [hunk 2]
```diff
@@ -86,6 +86,10 @@ private static void copyFile(Path source, Path target, boolean foreign, CopyOpti
             return;
         }
 
+        if (target.getFileSystem().provider().getScheme().equals("latch")) {
+            target.getFileSystem().provider().copy(source, target, options);
```
Comment (rahuldesai1) [inline modules/nf-commons/src/main/nextflow/file/CopyMoveHelper.java:90]: return here after copy?

File: modules/nf-httpfs/src/main/nextflow/file/http/LatchFileSystemProvider.groovy [hunk 3]
```diff
@@ -237,7 +237,15 @@ class LatchFileSystemProvider extends XFileSystemProvider {
     void delete(Path path) throws IOException {}
 
     @Override
-    void copy(Path source, Path target, CopyOption... options) throws IOException {}
+    void copy(Path source, Path target, CopyOption... options) throws IOException {
+        if (target.scheme == "latch" && source.scheme == "file") {
+            LatchPath lp = (LatchPath) target
+            lp.upload(source)
```
Comment (rahuldesai1) [inline modules/nf-httpfs/src/main/nextflow/file/http/LatchFileSystemProvider.groovy:243]: its probably fine but are we good to ignore the copy options? Like what happens if the file exists on the remote?

**Review by rahuldesai1 — CHANGES_REQUESTED** [review]

**Comment by rahuldesai1** [comment]
version + changelog plz

File: modules/nf-httpfs/src/main/nextflow/file/http/LatchFileSystemProvider.groovy (hunk 3 above)
Comment (ayushkamat) [reply modules/nf-httpfs/src/main/nextflow/file/http/LatchFileSystemProvider.groovy:243]: ill add a todo to check and add good errors for that + other failure modes like directory exists at path, etc

**Pull request merged**